numerical/area/bounds/_base.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from abc import ABCMeta, abstractmethod
+import matplotlib.pyplot as plt
+
+
+class BaseBoundary(metaclass=ABCMeta):
+    @abstractmethod
+    def plot(self, save_path=None):
+        """ Visualizes boundary. """
+
+    def _set_described_rect(self):
+        """ Set n-dim rectangular bound which describes initial bound. """
+
+    @abstractmethod
+    def _is_boundary_valid(self):
+        """ Validate the data which gives for boundary initialization. """
+
+
+class BoundaryIterable(metaclass=ABCMeta):
+    """ The class represents iterator interface for boundaries. """
+    def __init__(self, data: list):
+        self._data = data
+        self.bounds_count = len(data)
+
+    def __iter__(self):
+        self._bound_id = 0
+        return self
+
+    def __next__(self):
+        if self._bound_id == self.bounds_count:
+            raise StopIteration
+        self._bound_id += 1
+        return self._data[self._bound_id - 1]
```

Comment:
AFAIK `__next__` should return the next element, so why `self._data[self._bound_id - 1]` ?

Reply:
OK